docs/src/tutorials/diffusion_implicit_heat_equation.md

```diff
@@ -0,0 +1,186 @@
+
+with boundary conditions: ``∇T(z=a) = ∇T_{bottom}, T(z=b) = T_{top}``. We'll solve these equations on cell faces, rather than cell centers. The same exercise could easily be done on cell centers.
+
+First, we'll use / import some packages:
+
+```julia
+import Plots
+using LinearAlgebra
+using DiffEqBase
+using OrdinaryDiffEq: SplitODEProblem, solve, IMEXEuler
+import SciMLBase
+```
+
+Next, we'll define some global problem parameters:
+```julia
+a,b, n = 0, 1, 10               # zmin, zmax, number of cells
+n̂_min, n̂_max = -1, 1            # Outward facing unit vectors
+α = 100;                        # thermal diffusivity, larger means more stiff
+β, γ = 10000, π;                # source term coefficients
+Δt = 1000;                      # timestep size
+N_t = 10;                       # number of timesteps to take
+FT = Float64;                   # float type
+Δz = FT(b-a)/FT(n)
+Δz² = Δz^2;
+∇²_op = [1/Δz², -2/Δz², 1/Δz²]; # interior Laplacian operator
+∇T_bottom = 10;                 # Temperature gradient at the top
+T_top = 1;                      # Temperature at the bottom
+S(z) = β*sin(γ*z)               # source term, (sin for easy integration)
+zf = range(a, b, length=n+1);   # coordinates on cell faces
+```
+
+## Derivation of analytic solution
+Here, we'll derive the analytic solution:
+
+```math
+\frac{∂²T}{∂²z} = -S(z)/α = -β \sin(γ z)/α \\
+\frac{∂T}{∂z} = β \cos(γ z)/(γ α)+c_1 \\
+T(z) = β \sin(γ z)/(γ^2 α)+c_1 z+c_2, \qquad \text{(generic solution)}
+```
+Apply bottom boundary condition:
+```math
+\frac{∂T}{∂z}(a) = β \cos(γ a)/(γ α)+c_1 = ∇T_{bottom} \\
+c_1 = ∇T_{bottom}-β \cos(γ a)/(γ α)
+```
+
+Apply top boundary condition:
+```math
+T(b) = β \sin(γ b)/(γ^2 α)+c_1 b+c_2 = T_{top} \\
+c_2 = T_{top}-(β \sin(γ b)/(γ^2 α)+c_1 b)
+```
+
+And now let's define this in a julia function:
+```julia
+function T_analytic(z) # Analytic steady state solution
+    c1 = ∇T_bottom-β*cos(γ*a)/(γ*α)
+    c2 = T_top-(β*sin(γ*b)/(γ^2*α)+c1*b)
+    return β*sin(γ*z)/(γ^2*α)+c1*z+c2
+end
+```
+
+Here, we'll derivation the matrix form of the temporal discretization we wish to use (diffusion-implicit and explicit Euler):
+```math
+∂_t T = α ∇²T + S \\
+(T^{n+1}-T^n) = Δt (α  ∇²T^{n+1} + S) \\
+(T^{n+1} - Δt α ∇²T^{n+1}) = T^n + Δt S \\
+(I - Δt α ∇²) T^{n+1} = T^n + Δt S
+```
+
+Using the following index denotion:
+
+ - `i` first interior index
+ - `b` boundary index
+ - `g` ghost index
+
+we'll derive the Dirichlet boundary stencil & source:
+```math
+∂_t T = α (T[i-1]+T[b]-2 T[i])/Δz² + S \\
+∂_t T = α (T[i-1]-2 T[i])/Δz² + S + α T[b] / Δz²
+```
+
+and Neumann boundary stencil & source:
+```math
+∇T_{bottom} n̂ = (T[g] - T[i])/(2Δz), \qquad    n̂ = [-1,1] ∈ [zmin,zmax] \\
+T[i] + 2 Δz ∇T_{bottom} n̂ = T[g] \\
+∂_t T = α (((T[i] + 2 Δz ∇T_{bottom} n̂) - T[b])/Δz - (T[b] - T[i])/Δz)/Δz + S \\
+∂_t T = α (((T[i]) - T[b])/Δz - (T[b] - T[i])/Δz)/Δz + S + α 2 Δz ∇T_{bottom} n̂/Δz² \\
+∂_t T = α (2 T[i] - 2 T[b])/Δz² + S + 2α ∇T_{bottom} n̂/Δz
+```
```

Comment:
Yeah, all of this is too much build up and loses the message. I think this should all go to https://github.com/SciML/SciMLTutorials.jl . It would be good to get an IMEX in here, but such a tutorial would want to have no more than a 1 paragraph problem statement, which is very different from what this is trying to do. This is trying to show a workflow specific to a use cases, which is what the extended tutorial set is made for.